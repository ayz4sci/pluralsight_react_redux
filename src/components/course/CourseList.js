import React, {PropTypes} from 'react';
import CourseListRow from './CourseListRow';

const CourseList = ({courses, sort, deleteCourse}) => {
    return (
        <table className="table">
            <thead>
                <tr>
                    <th>&nbsp;</th>
                     {/* eslint-disable-next-line */}
                    <th><a href="#" onClick={sort}>Title</a></th>
                    <th>Author</th>
                    <th>Category</th>
                    <th>Length</th>
                    <th>&nbsp;</th>
                </tr>
            </thead>
            <tbody>
                { courses.map(course => 
                    <CourseListRow 
                        key={course.id} 
                        course={course}
                        deleteCourse={deleteCourse} />
                )}
            </tbody>
        </table>
    );
};

CourseList.propTypes = {
    courses: PropTypes.array.isRequired,
    sort: PropTypes.func.isRequired
};

export default CourseList;