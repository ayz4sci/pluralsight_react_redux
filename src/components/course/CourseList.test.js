import React from 'react';
import renderer from 'react-test-renderer';
import CourseList from './CourseList';

describe ('CourseList Test', () => {
    it('renders correctly using snapshot', () => {
        const props = {
            courses: [{id: 'A', watchHref: 'B', title: 'C', authorId: 'D', length: 'E', category: 'F'}],
            deleteCourse: () => Promise.resolve() 
        };
        const tree = renderer.create(<CourseList {...props} />).toJSON();
        expect(tree).toMatchSnapshot();
    })
});