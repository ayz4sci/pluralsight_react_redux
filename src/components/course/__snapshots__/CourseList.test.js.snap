// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseList Test renders correctly using snapshot 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <th>
         
      </th>
      <th>
        Title
      </th>
      <th>
        Author
      </th>
      <th>
        Category
      </th>
      <th>
        Length
      </th>
      <th>
         
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        <a
          href="B"
          rel="noopener noreferrer"
          target="_blank"
        >
          Watch
        </a>
      </td>
      <td>
        <a
          onClick={[Function]}
          style={Object {}}
        >
          C
        </a>
      </td>
      <td>
        D
      </td>
      <td>
        F
      </td>
      <td>
        E
      </td>
      <td>
        <input
          className="btn btn-danger btn-sm"
          onClick={[Function]}
          type="submit"
          value="Delete"
        />
      </td>
    </tr>
  </tbody>
</table>
`;
